cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -299,28 +299,44 @@ func (d *Daemon) mustInitializeStorage(cfg *config.Config) *feewindow.FeeWindows
 	defer cancelReadTxMeta()
 	var initialSeq uint32
 	var currentSeq uint32
-	dataMigrations, err := db.BuildMigrations(readTxMetaCtx, d.logger, d.db, cfg)
+	applicableRange, err := db.GetMigrationLedgerRange(readTxMetaCtx, d.db, cfg.HistoryRetentionWindow)
 	if err != nil {
-		d.logger.WithError(err).Fatal("could not build migrations")
+		d.logger.WithError(err).Fatal("could not get ledger range for migration")
 	}
 
-	// Merge migrations range and fee stats range to get the applicable range
-	latestLedger, err := db.NewLedgerEntryReader(d.db).GetLatestLedgerSequence(readTxMetaCtx)
+	maxFeeRetentionWindow := max(cfg.ClassicFeeStatsLedgerRetentionWindow, cfg.SorobanFeeStatsLedgerRetentionWindow)
+	feeStatsRange, err := db.GetMigrationLedgerRange(readTxMetaCtx, d.db, maxFeeRetentionWindow)
 	if err != nil {
-		d.logger.WithError(err).Fatal("failed to get latest ledger sequence: %w", err)
+		d.logger.WithError(err).Fatal("could not get ledger range for fee stats")
 	}
 
-	maxFeeRetentionWindow := max(cfg.ClassicFeeStatsLedgerRetentionWindow, cfg.SorobanFeeStatsLedgerRetentionWindow)
-	ledgerSeqRange := &db.LedgerSeqRange{FirstLedgerSeq: latestLedger - maxFeeRetentionWindow, LastLedgerSeq: latestLedger}
-	applicableRange := dataMigrations.ApplicableRange()
-	ledgerSeqRange = ledgerSeqRange.Merge(applicableRange)
+	// Combine the ledger range for fees, events and transactions
+	ledgerSeqRange := feeStatsRange.Merge(applicableRange)
+
+	// Start a common db transaction for the entire migration duration
+	err = d.db.Begin(readTxMetaCtx)
```

Comment:
I understand that a common migration needs to be used for all migrations. But it can be _hidden_ inside the migration code (my guess is multimigration would be the best place to place it). It would make this code more readable and would most probably make rolling back/commiting much simpler.

Reply:
@2opremio Yeah I agree. Nice suggestion